src/plugins/data/server/search/aggs/aggs_service.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  AggsCommonService,
+  AggTypesRegistry,
+  aggsRequiredUiSettings,
+  calculateBounds,
+  getAggTypes,
+  TimeRange,
+} from '../../../common';
+import { FieldFormatsStart } from '../../field_formats';
+import { AggsSetup, AggsStart } from './types';
+
+/** @internal */
+export interface AggsSetupDependencies {
+  registerFunction: ExpressionsServiceSetup['registerFunction'];
+}
+
+/** @internal */
+export interface AggsStartDependencies {
+  fieldFormats: FieldFormatsStart;
+  uiSettings: UiSettingsServiceStart;
+}
+
+/**
+ * The aggs service provides a means of modeling and manipulating the various
+ * Elasticsearch aggregations supported by Kibana, providing the ability to
+ * output the correct DSL when you are ready to send your request to ES.
+ */
+export class AggsService {
+  private registerFunction?: ExpressionsServiceSetup['registerFunction'];
+  /**
+   * This is a temporary cache for agg types which are registered by other
+   * plugins in `setup`. Its contents are merged with the default agg types
+   * and added to the "real" registry which is instantiated when someone calls
+   * `data.search.aggs.asScopedToClient`.
+   */
+  private readonly aggTypesCache = new AggTypesRegistry();
+
+  /**
+   * getForceNow uses window.location on the client, so we must have a
+   * separate implementation of calculateBounds on the server.
+   */
+  private calculateBounds = (timeRange: TimeRange) => calculateBounds(timeRange, {});
+
+  public setup({ registerFunction }: AggsSetupDependencies): AggsSetup {
+    this.registerFunction = registerFunction;
+
+    return {
+      types: this.aggTypesCache.setup(),
+    };
+  }
+
+  public start({ fieldFormats, uiSettings }: AggsStartDependencies): AggsStart {
+    return {
+      asScopedToClient: async (savedObjectsClient: SavedObjectsClientContract) => {
+        const uiSettingsClient = uiSettings.asScopedToClient(savedObjectsClient);
+        const formats = await fieldFormats.fieldFormatServiceFactory(uiSettingsClient);
+
+        // cache ui settings, only including items which are explicitly needed by aggs
+        const uiSettingsCache = pick(await uiSettingsClient.getAll(), aggsRequiredUiSettings);
+        const getConfig = <T = any>(key: string): T => {
+          return uiSettingsCache[key];
+        };
+
+        const defaultAggTypes = getAggTypes({
+          calculateBounds: this.calculateBounds,
+          getConfig,
+          getFieldFormatsStart: () => ({
+            deserialize: formats.deserialize,
+            getDefaultInstance: formats.getDefaultInstance,
+          }),
+          /**
+           * Date histogram and date range need to know whether we are using the
+           * default timezone, but `isDefault` is not currently offered on the
+           * server, so we need to manually check for the default value.
+           */
+          isDefaultTimezone: () => getConfig('dateFormat:tz') === 'Browser',
```

Comment:
Ideally we'd add `isDefault` to the `uiSettings` service on the server so that we don't need to manually check against the default value. I'm looking into this as an alternative.